### ISSUE TITLE
Is it possible to provide a URI to create a PKey? 

### ISSUE THREAD

**Issue by WolfDan** [sfackler/rust-openssl#2059]
I'm working with the pkcs11-provider for OpenSSL 3; one way to use the provider is via a URI; openssl internally will make use of `OSSL_STORE_open_ex` to get the provider key and use it internally. I can't see any function on `PKey` or a store module, so I'm wondering if that's possible with the library

For example, this command works with OpenSSL, but I'm unsure if `-key "pkcs11:type=private;id=%00%01"` will translate into rust-openssl

`openssl req -new -x509 -days 365 -subj '/CN=my key/' -sha256 -provider pkcs11 -key "pkcs11:type=private;id=%00%01" -out cert.pem`

Thank you!

**Comment by sfackler** [comment]
I don't think we currently expose anything like that but I'd be happy to take a PR.

**Comment by Firstyear** [comment]
Similar to this, a method to create keys into a PKCS11 provider would probably be good here to complete the lifecycle. The challenge might be the provider setup if you have keys coming from different sources though.

**Comment by Firstyear** [comment]
@WolfDan I'm going to give this a go.
